File: legacy/chips/busterup.py
# -*- coding: utf-8 -*-
# This file is part of MMBN Online
# MMBN Online is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# MMBN Online is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# You should have received a copy of the GNU General Public License
# along with MMBN Online.  If not, see <http://www.gnu.org/licenses/>.

# Copyright (C) 2008-2010 Chris Santiago and Brandon Evans.
# http://mmbnonline.net/

"""A version of the Chip class."""

from chip import Chip as Parent

__all__ = ['Chip']

class Chip(Parent):
    """A version of the Chip class."""
    def properties(self):
        """Overwrite the default properties."""
        self.codes = ('*',)
        self.description = 'Power goes up by 1'
        self.element = 'plus'
        self.name = 'Buster Up'
        self.plus = 1
        self.short = 'BusterUp'
        self.stars = 2

    def use(self):
        self.owner.power += self.plus

File: server/chips/busterUp.js
'use strict';

var Chip = require(__dirname + '/'),
    extend = require(__dirname + '/extend');

function BusterUpConstructor(io, config, game, character) {
    var self = this;

    this.properties.codes = ['*'];
    this.properties.description = 'Power goes up by 1';
    this.properties.element = 'plus';
    this.properties.name = 'Buster Up';
    this.properties.short = 'BusterUp';
    this.properties.plus = 1;
    this.properties.stars = 2;

    this.use = function use() {
        character.boostBusterPower(self.properties.plus);
    };
}

module.exports = function BusterUp(io, config, game, character) {
    return extend(BusterUpConstructor, Chip, io, config, game, character);
};
